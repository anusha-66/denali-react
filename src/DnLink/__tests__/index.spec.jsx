/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnLink, DnIcon } from '../../index'; // src/index.tsx

describe('DnLink Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink>Home</DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink isActive isDisabled isSmall isSecondary isSub>
                    Home
                </DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at front', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink frontIconComponent={<DnIcon icon="add-circle" />}>Home</DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at front with additional props', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink isActive isDisabled isSmall isSecondary isSub frontIconComponent={<DnIcon icon="add-circle" />}>
                    Home
                </DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at back', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink backIconComponent={<DnIcon icon="external" className="is-sub" />}>Home</DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at back with all props', () => {
        expect.assertions(1);

        const rendered = render(
            <a href="#home">
                <DnLink
                    isActive
                    isDisabled
                    isSmall
                    isSecondary
                    isSub
                    backIconComponent={<DnIcon icon="external" className="is-sub" />}
                >
                    Home
                </DnLink>
            </a>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
