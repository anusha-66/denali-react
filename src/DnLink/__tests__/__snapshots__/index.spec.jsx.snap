// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnLink Tests should render correctly with Icon at back 1`] = `
<a
  href="#home"
>
  <span
    class="link has-icon-back"
  >
    Home
    <i
      class="d-icon d-external is-default is-sub"
    />
  </span>
</a>
`;

exports[`DnLink Tests should render correctly with Icon at back with all props 1`] = `
<a
  href="#home"
>
  <span
    class="link is-active is-disabled is-small is-secondary is-sub has-icon-back"
  >
    Home
    <i
      class="d-icon d-external is-default is-sub"
    />
  </span>
</a>
`;

exports[`DnLink Tests should render correctly with Icon at front 1`] = `
<a
  href="#home"
>
  <span
    class="link has-icon-front"
  >
    <i
      class="d-icon d-add-circle is-default"
    />
    Home
  </span>
</a>
`;

exports[`DnLink Tests should render correctly with Icon at front with additional props 1`] = `
<a
  href="#home"
>
  <span
    class="link is-active is-disabled is-small is-secondary is-sub has-icon-front"
  >
    <i
      class="d-icon d-add-circle is-default"
    />
    Home
  </span>
</a>
`;

exports[`DnLink Tests should render correctly with custom props 1`] = `
<a
  href="#home"
>
  <span
    class="link is-active is-disabled is-small is-secondary is-sub"
  >
    Home
  </span>
</a>
`;

exports[`DnLink Tests should render with default props 1`] = `
<a
  href="#home"
>
  <span
    class="link"
  >
    Home
  </span>
</a>
`;
